src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,151 @@
+package duke.parser;//package duke;
+
+import duke.storage.Storage;
+import duke.task.*;
+import duke.ui.Ui;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.regex.Pattern;
+
+/**
+ * The Parser class parses user input and executes corresponding commands.
+ */
+public class Parser {
+    Ui ui;
+    Storage storage;
+    TaskList taskList;
+
+    /**
+     * Constructs a Parser object with the specified Ui, Storage, and TaskList.
+     *
+     * @param ui        the user interface
+     * @param storage   the storage handler
+     * @param taskList  the task list
+     */
+    public Parser(Ui ui, Storage storage, TaskList taskList) {
+        this.ui = ui;
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Validates if a given string represents a valid date in the format "yyyy-MM-dd".
+     *
+     * @param dateString the string representation of the date
+     * @return true if the string is a valid date, false otherwise
+     */
+    private static boolean isValidDate(String dateString) {
```

Comment:
Maybe you can name your method as a verb (e.g. checkValidity) ?

Reply:
Actually I think isValidDate is fine since this is a boolean method. Adheres to the Java coding standard. 👍